tools/dm/cluster-operations.md

```diff
@@ -0,0 +1,396 @@
+
+      要解决这个问题，请按照[手动处理 Sharding DDL Lock](/tools/dm/manually-handling-sharding-ddl-locks.md#场景二-unlock-过程中部分-dm-worker-重启) 中描述的步骤操作。
+
+**总结**：尽量避免在 sharding DDL 同步过程中重启 DM-worker。
+
+#### DM-master 重启事项
+
+由 DM-master 维护的信息包括以下两种。重启 DM-master 不会持久化保存这些信息的相关数据。
+
+- 任务与 DM-master 之间的对应关系
+- Sharding DDL lock 的相关信息
+
+DM-master 重启时会自动向每个 DM-worker 实例请求任务信息，重建任务与 DM-worker 之间的对应关系，并从每个 DM-worker 实例获取 sharding DDL 信息。这样，就可以准确重建相应的 DDL lock，也可以自动解除 sharding DDL lock。
+
+### 重启 DM-worker
+
+> **注意**：尽量避免在 sharding DDL 同步过程中重启 DM-worker。
+
+使用以下两种方法中任一种重启 DM-worker 组件：
+
+- 对 DM-worker 执行滚动升级。
+
+    ```bash
+    $ ansible-playbook rolling_update.yml --tags=dm-worker
+    ```
+
+- 先停止 DM-worker，然后重启。
+
+    ```bash
+    $ ansible-playbook stop.yml --tags=dm-worker
+    $ ansible-playbook start.yml --tags=dm-worker
+    ```
+
+### 重启 DM-master
+
+在以下两种方法中任选一种，重启 DM-master 组件：
+
+- 对 DM-master 执行滚动升级。
+
+    ```bash
+    $ ansible-playbook rolling_update.yml --tags=dm-master
+    ```
+
+- 停止 DM-master，然后重启。
+
+    ```bash
+    $ ansible-playbook stop.yml --tags=dm-master
+    $ ansible-playbook start.yml --tags=dm-master
+    ```
+
+## 更新组件版本
+
+1. 下载 DM 二进制文件。
+
+    1. 从 `downloads` 目录删除已有文件。 
+
+        ```
+        $ cd /home/tidb/dm-ansible
+        $ rm -rf downloads
+        ```
+
+    2. 用 Playbook 下载最新 DM 二进制文件。这会自动替换 `/home/tidb/dm-ansible/resource/bin/` 中已有文件。 
+
+        ```
+        $ ansible-playbook local_prepare.yml
+        ```
+
+2. 使用 Ansible 执行滚动升级。
+
+    1. 对 DM-worker 实例执行滚动升级：
+
+        ```
+        ansible-playbook rolling_update.yml --tags=dm-worker
+        ```
+
+    2. 对 DM-master 实例执行滚动升级：
+
+        ```
+        ansible-playbook rolling_update.yml --tags=dm-master
+        ```
+
+    3. 升级 dmctl：
+
+        ```
+        ansible-playbook rolling_update.yml --tags=dmctl
+        ```
+
+    4. 对 DM-worker， DM-master， 以及 dmctl 整体执行滚动升级：
+
+        ```
+        ansible-playbook rolling_update.yml
+        ```
+
+## 创建 DM-worker 实例
+
+假设您想要在机器 `172.16.10.74` 上创建一个名为 `dm_worker3` 的 DM-worker 实例，按以下步骤操作：
+
+1. 为中控机设置 SSH 互信以及 sudo 规则。
+
+    1. 参考[在中控机上配置 SSH 互信和 sudo 规则](/tools/dm/deployment.md#第-5-步在中控机上配置-SSH-互信和-sudo-规则)，使用 `tidb` 用户登陆至中控机，并将 `172.16.10.74` 添加至 `hosts.ini` 文件中的 `[servers]` 部分。
+
+        ```
+        $ cd /home/tidb/dm-ansible
+        $ vi hosts.ini
+        [servers]
+        172.16.10.74
+
+        [all:vars]
+        username = tidb
+        ```
+
+    2. 运行以下命令。根据屏幕提示，输入 `root` 用户密码以部署 `172.16.10.74`。
+
+        ```
+        $ ansible-playbook -i hosts.ini create_users.yml -u root -k
+        ```
+
+        该步在 `172.16.10.74` 机器上创建了一个 `tidb` 用户，设置了 sudo 规则，并为中控机与该机器配置了 SSH 互信。
+
+2. 修改 `inventory.ini` 文件，创建新 DM-worker 实例 `dm_worker3`。
+
+    ```
+    [dm_worker_servers]
+    dm_worker1 source_id: "instance-1" ansible_host=172.16.10.72 server_id=101 mysql_host=172.16.10.81 mysql_user=root mysql_password='VjX8cEeTX+qcvZ3bPaO4h0C80pe/1aU=' mysql_port=3306
```

Comment:
source_id: "instance-1" Is this the right format her? or we should write it like `source_id="instance-1"` @GregoryIan